### PR TITLE
Display registration dialogs

### PR DESCRIPTION
**Pull request by settinghead**
image
image
image
image

### CODE REVIEW

**Comment by settinghead** [comment]
@alex-deaconu @ezequielc the JS implementation in this PR is ready. I am still writing unit tests, which will come in shortly as a new commit. Meanwhile feel free to validate the logic of the new implementation. Thanks.

File: src-js/Viewer/Data/DisplayRegistrationController.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  var url = Global.DATA_SERVER_URL.replace('{0}', 'display');
+  url = url.replace('{1}', displayId);
+  url += '?callback=?';
+  _callUrlNative(url, MODE_VALIDATE, cb);
+};
+
+
+function _callUrlNative(url, mode, cb) {
+  try {
+    window.writeToLog('Register Display request - start');
```
Comment (alex-deaconu) [inline src-js/Viewer/Data/DisplayRegistrationController.js]: @settinghead just a quick note, we should be using `Logger.log` here. Thanks.

File: src-js/Viewer/Data/ViewerDataController.js [hunk 2]
```diff
@@ -125,97 +123,88 @@ var _setNewViewerData = function (newViewerData) {
...
-  catch (err) {
-    Logger.log('getParentData - ' + id + ' - ' + err.message);
-  }
+  return undefined;
 };
 
-var _getPollInterval = function () {
-  return viewerData &&
+var _getPollInterval = () =>
+  (viewerData &&
```
Comment (alex-deaconu) [inline src-js/Viewer/Data/ViewerDataController.js]: @settinghead is this es6? or are you missing the `return`? Thanks.

File: src-js/Viewer/Data/ViewerDataController.js [hunk 3]
```diff
@@ -125,97 +123,88 @@ var _setNewViewerData = function (newViewerData) {
...
-  return viewerData ? viewerData.signature : '';
-};
+service.getSig = () => (viewerData ? viewerData.signature : '');
 
-service.getEmbedItemData = function (id, parentId) {
+service.getEmbedItemData = (id, parentId) => {
   // first look for the presentation
   if (id && viewerData && viewerData.content && viewerData.content.presentations) {
-    for (var i = 0; i < viewerData.content.presentations.length; i++) {
+    for (var i = 0; i < viewerData.content.presentations.length; i += 1) {
```
Comment (alex-deaconu) [inline src-js/Viewer/Data/ViewerDataController.js]: @settinghead es6 notation here? Thanks.

File: src-js/Viewer/Data/ViewerDataController.js (hunk 2 above)
Comment (settinghead) [inline src-js/Viewer/Data/ViewerDataController.js]: @alex-deaconu yeah this is ES6, and return is implicit if there are no curly braces that surround a function.

File: src-js/Viewer/Data/ViewerDataController.js [hunk 4]
```diff
@@ -125,97 +123,88 @@ var _setNewViewerData = function (newViewerData) {
...
   if (!pollingTimer) {
-    if (interval == 0) {
-      interval = _getPollInterval();
-    }
-    else {
-      interval += 2;
+    if (interval === 0) {
+      newInterval = _getPollInterval();
+    } else {
+      newInterval += 2;
```
Comment (settinghead) [inline src-js/Viewer/Data/ViewerDataController.js]: @alex-deaconu good find. The creation of new variable is to comply with linting rule (function parameter shouldn't be overridden). but we can turn it off for now.

File: src-js/Viewer/Data/ViewerDataController.js (hunk 3 above)
Comment (settinghead) [inline src-js/Viewer/Data/ViewerDataController.js]: @alex-deaconu it's one of a set of recommended rule I'm following:
http://eslint.org/docs/rules/no-plusplus

File: src-js/Viewer/Data/ViewerDataController.js (hunk 4 above)
Comment (settinghead) [inline src-js/Viewer/Data/ViewerDataController.js]: @alex-deaconu rationale is here:
http://eslint.org/docs/rules/no-param-reassign

File: src-js/Viewer/Data/ViewerDataController.js (hunk 2 above)
Comment (alex-deaconu) [inline src-js/Viewer/Data/ViewerDataController.js]: @settinghead 👍 thanks for the clarification.

File: src-js/Viewer/Data/ViewerDataController.js (hunk 3 above)
Comment (alex-deaconu) [inline src-js/Viewer/Data/ViewerDataController.js]: @settinghead I'm good either way. However, we need to be consistent. Thanks.

File: src-js/Viewer/Data/ViewerDataController.js (hunk 4 above)
Comment (alex-deaconu) [inline src-js/Viewer/Data/ViewerDataController.js]: @settinghead I'm good either way. Thanks.

**Comment by settinghead** [comment]
@alex-deaconu all fixed & unit tests added.

**Comment by alex-deaconu** [comment]
@settinghead just testing this out, there's an error when accessing:
http://display-registration-dot-rvaviewer-test.appspot.com/?type=display&id=asdfasdf&player=true

`Cannot read property 'getMessage' of undefined`

Basically, when entering an Invalid Display Id usecase. Can you take a look? Thanks.

**Comment by alex-deaconu** [comment]
@settinghead also a slight UX misalignment issue with the close button here: http://screencast.com/t/IYfAzNizhdka. And there's a thin line appearing across the top, not sure where that's from. Thanks.

File: src-js/Viewer/Data/DisplayRegistrationController.js [hunk 5]
```diff
@@ -35,16 +35,15 @@ service.validateDisplayId = (displayId, cb) => {
...
-
     $.getJSON(url,
     { format: 'json' },
     (result) => {
-      try {
-        Logger.log('Register Display response - active');
-        _processResponse(result, mode, cb);
-      } catch (err) {
-        Logger.log(`Register Display request failed - ${url} - ${err.message}`);
-      }
```
Comment (alex-deaconu) [inline src-js/Viewer/Data/DisplayRegistrationController.js]: @settinghead wondering why this is commented out now. Thanks.

Comment (settinghead) [inline src-js/Viewer/Data/DisplayRegistrationController.js]: @alex-deaconu doh!

**Comment by settinghead** [comment]
@alex-deaconu all 3 problems should be fixed.

**Comment by alex-deaconu** [comment]
@settinghead looks good 👍 . Wondering if you can squash, but hold off on merging until QA is finished testing this version. Thanks.

**Pull request merged**